t/t2803-flux-pstree.t

```diff
@@ -0,0 +1,285 @@
+'
+test_expect_success 'flux-pstree -x --skip-root=no works in empty instance' '
+	name="empty-extended-no-skip-root" &&
+	flux pstree -x --skip-root=no > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	test $(cat ${name}.output | wc -l) -eq 2
+'
+#  Start a child instance that immediately exits, so that we can test
+#   that `flux pstree` doesn't error on child instances that are no
+#   longer running.
+#
+#  Then, start a job hiearachy with 2 child instances, each of which
+#   run a sleep job, touch a ready.<id> file, then block waiting for
+#   the sleep job to finish.
+#
+test_expect_success 'start a recursive job' '
+	id=$(flux mini submit flux start /bin/true) &&
+	rid=$(flux mini submit -n2 \
+		flux start \
+		flux mini submit --wait --cc=1-2 flux start \
+			"flux mini submit sleep inf && \
+			 touch ready.\$FLUX_JOB_CC && \
+			 flux queue idle") &&
+	flux job wait-event $id clean
+'
+test_expect_success 'wait for hierarchy to be ready' '
+	flux getattr broker.pid &&
+	$waitfile -t 60 ready.1 &&
+	$waitfile -t 60 ready.2
+'
+test_expect_success 'flux-pstree works' '
+	name="normal" &&
+	flux pstree > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	.
+	└── flux
+	    ├── flux
+	    │   └── sleep
+	    └── flux
+	        └── sleep
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
+test_expect_success 'flux-pstree JOBID works' '
+	name="jobid" &&
+	flux pstree $rid > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	flux
+	├── flux
+	│   └── sleep
+	└── flux
+	    └── sleep
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
+test_expect_success 'flux-pstree works when run inside child job' '
+	name="proxy" &&
+	flux proxy $rid flux pstree > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	flux
+	├── flux
+	│   └── sleep
+	└── flux
+	    └── sleep
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
+
+test_expect_success 'flux-pstree -a works' '
+	name="all" &&
+	flux pstree -a > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	.
+	├── flux
+	│   ├── flux
+	│   │   └── sleep:R
+	│   └── flux
+	│       └── sleep:R
+	└── flux:CD
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
+test_expect_success 'flux-pstree truncates at COLUMNS' '
+	name="truncated" &&
+	COLUMNS=16 flux pstree -a > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	.
+	├── flux
+	│   ├── flux
+	│   │   └── sle+
+	│   └── flux
+	│       └── sle+
+	└── flux:CD
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
+test_expect_success 'flux-pstree truncates at COLUMNS' '
+	name="truncated" &&
+	COLUMNS=16 flux pstree -a > ${name}.output &&
+	test_debug "cat ${name}.output" &&
+	cat <<-EOF >${name}.expected &&
+	.
+	├── flux
+	│   ├── flux
+	│   │   └── sle+
+	│   └── flux
+	│       └── sle+
+	└── flux:CD
+	EOF
+	test_cmp ${name}.expected ${name}.output
+'
```

Comment:
duplicated test? (or if its different and i'm missing the difference, the title of the test is the same atleast)

Reply:
Good catch. Accidental cut-and-paste. Will fix.